bspump/declarative/expression/string/joinexpr.py

```diff
@@ -1,37 +1,59 @@
-from ...abc import Expression, evaluate
+from ...abc import Expression
 from ..value.valueexpr import VALUE
 
 
 class JOIN(Expression):
 
 	Attributes = {
-		"Items": [],  # TODO: This ...
 		"Char": ["*"],  # TODO: This ...
 		"Miss": ["*"],  # TODO: This ...
 	}
 
 	def __init__(self, app, *, arg_items, arg_delimiter=" ", arg_miss=""):
 		super().__init__(app)
 		self.App = app
+
 		self.Items = arg_items
+		self.ItemsNormalized = []
+
 		self.Char = arg_delimiter
 
 		if not isinstance(arg_miss, Expression):
 			self.Miss = VALUE(app, value=arg_miss)
 		else:
 			self.Miss = arg_miss
 
+	def get_outlet_type(self):
+		return str.__name__
+
+	def consult_inlet_type(self, key, child):
+		return str.__name__
+
+	def set(self, key, value):
```

Comment:
?

Reply:
@ateska The same as above. Expressions in lists or dicts do not pass optimization.